src/parse.y

```diff
@@ -165,6 +165,7 @@ transtype(A) ::= .             {A = TK_DEFERRED;}
 transtype(A) ::= DEFERRED(X).  {A = @X; /*A-overwrites-X*/}
 transtype(A) ::= IMMEDIATE(X). {A = @X; /*A-overwrites-X*/}
 transtype(A) ::= EXCLUSIVE(X). {A = @X; /*A-overwrites-X*/}
+transtype(A) ::= READONLY.     {A = TK_IMMEDIATE; /*A-overwrites-X*/}
```

Comment:
We need at least a huge `FIXME:` here, since we want to keep libsql an independent library that's a drop-in replacement for SQLite. In the `FIXME` we should explain what will be implemented as `BEGIN READONLY` in the future (more on that later). `BEGIN READONLY` being transiently parsed as `BEGIN IMMEDIATE` is a hack we need for sqld, and it's fine, but in general `BEGIN READONLY` is the exact opposite of `BEGIN IMMEDIATE` :innocent: Doesn't `A = TK_DEFERRED` make way more sense?

In general, later we can add semantics for `BEGIN READONLY` to libSQL anyway - it's a transaction that only accepts readonly statements and errors out on its first attempt to take a write lock.

Reply:
Done!